frontend/src/hooks/staffing/useConsultantsFilter.ts

```diff
@@ -69,3 +80,81 @@ function inYearRanges(consultant: Consultant, yearRanges: YearRange[]) {
   }
   return false;
 }
+
+interface WeeklyTotal {
+  weeklyTotalBillable: Map<number, number>;
+  weeklyTotalBillableAndOffered: Map<number, number>;
+}
+
+function setWeeklyTotalBillable(
+  filteredConsultants: Consultant[],
+): WeeklyTotal {
+  const weeklyTotalBillable = new Map<number, number>();
+  const weeklyTotalBillableAndOffered = new Map<number, number>();
+
+  filteredConsultants.forEach((consultant) => {
+    consultant.bookings.forEach((booking) => {
+      if (weeklyTotalBillable.has(booking.weekNumber)) {
+        weeklyTotalBillable.set(
+          booking.weekNumber,
+          (weeklyTotalBillable.get(booking.weekNumber) || 0) +
+            booking.bookingModel.totalBillable,
+        );
+        weeklyTotalBillableAndOffered.set(
+          booking.weekNumber,
+          (weeklyTotalBillableAndOffered.get(booking.weekNumber) || 0) +
+            booking.bookingModel.totalBillable +
+            booking.bookingModel.totalOffered,
+        );
+      } else {
+        weeklyTotalBillable.set(
+          booking.weekNumber,
+          booking.bookingModel.totalBillable,
+        );
+        weeklyTotalBillableAndOffered.set(
+          booking.weekNumber,
+          booking.bookingModel.totalBillable +
+            booking.bookingModel.totalOffered,
+        );
+      }
+    });
+  });
+
+  return {
+    weeklyTotalBillable,
+    weeklyTotalBillableAndOffered,
+  };
+}
+
+function setWeeklyInvoiceRate(
+  filteredConsultants: Consultant[],
+  weeklyTotalBillable: Map<number, number>,
+) {
+  const weeklyInvoiceRate = new Map<number, number>();
+
+  weeklyTotalBillable.forEach((totalBillable, weekNumber) => {
+    let totalAvailableWeekHours = 0;
+
+    filteredConsultants.forEach((consultant) => {
+      consultant.bookings.forEach((booking) => {
+        if (booking.weekNumber === weekNumber) {
+          let consultantAvailableWeekHours =
+            37.5 -
```

Comment:
Aiai! Denne vil jo være 40t for sverige. Har vi noen sjekker for dette mot backend? Et kall etc?

Reply:
Kan jo ta en global search om det er mye hardkodet "37.5" rundt å gå :)

Reply:
Vi har all denne logikken backend, så vi kan gjerne gjøre litt backend-for-frontend og heller regne ut dette i API-kallet :)

Reply:
Men da må du regne ut alt på nytt igjen når du søker 🫠

Reply:
Aka. Nytt api kall pr bokstav man trykker, eller filter man velger

Reply:
Veldig godt poeng! Jeg tror kanskje utregningen bør gjøres på frontend, siden den er basert på konsulentene i filtreringen, så hvis det skal regnes ut backend må kanskje den filtrerte listen over konsulenter sendes frem og tilbake, noe som blir kostbart mtp filtrering som skjer for eksempel hver gang en bokstav skrives inn? 😅 Men variabelen for antall timer i uka, bør nok være i backend

Reply:
Enig, siden det påvirkes av filtre er det naturlig at det regnes frontend. 

Kommenterte et par ting som kan forbedres i det nye endepunktet da, men alt i alt veldig bra jobba :)